src/components/AudioButton/AudioButton.js

```diff
@@ -0,0 +1,45 @@
+import styled from 'styled-components/macro'
+import PropTypes from 'prop-types'
+import { ReactComponent as PlayIcon } from '../../assets/play.svg'
+import { ReactComponent as PauseIcon } from '../../assets/pause.svg'
+import useAudio from './useAudio'
+
+AudioButton.propTypes = {
+  songUrl: PropTypes.string,
+  id: PropTypes.string,
+  currentSong: PropTypes.string,
+  setCurrentSong: PropTypes.func,
+}
+
+export default function AudioButton({
+  songUrl,
+  id,
+  currentSong,
+  setCurrentSong,
+}) {
+  const { isSongPlaying, setIsSongPlaying } = useAudio({
+    songUrl,
+    id,
+    currentSong,
+  })
```

Comment:
Wow! I think I've never seen it like this - importing and exporting from and into a custom hook. Took me a while to understand but I think I like it :))